backend/application/syson-sysml-export/src/main/java/org/eclipse/syson/sysml/export/SysMLElementSerializer.java

```diff
@@ -124,7 +147,169 @@ public String caseDefinition(Definition def) {
+
+        return builder.toString();
+    }
+
+    private void appendFeatureSpecilizationPart(Appender builder, Feature feature) {
+        List<Redefinition> ownedRedefinition = feature.getOwnedRedefinition();
+        appendRedefinition(builder, ownedRedefinition, feature);
+
+        appendReferenceSubsetting(builder, feature.getOwnedReferenceSubsetting(), feature);
+
+        List<Subsetting> ownedSubsetting = new ArrayList<>(feature.getOwnedSubsetting());
+        ownedSubsetting.removeAll(ownedRedefinition);
+        ownedSubsetting.remove(feature.getOwnedReferenceSubsetting());
+
+        appendSubsettings(builder, ownedSubsetting, feature);
+
+        appendFeatureTyping(builder, feature.getOwnedTyping(), feature);
+        this.appendMultiplicityPart(builder, feature);
+    }
+
+    private void appendSubsettings(Appender builder, List<Subsetting> subSettings, Element element) {
+        if (!subSettings.isEmpty()) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.SUBSETTING);
+            builder.appendSpaceIfNeeded().append(subSettings.stream().map(getSubsettedFeature())
+                    .filter(Objects::nonNull)
+                    .map(superFeature -> getDeresolvableName(superFeature, element))
+                    .collect(Collectors.joining(", ")));
+        }
+
+    }
+
+    // TODO : when the function Subsetting::getSubsettedFeature() is correctly implemented then remove this function and
+    // use only Subsetting::getSubsettedFeature() in appendSubsettings();
+    private Function<? super Subsetting, ? extends Feature> getSubsettedFeature() {
+        return subsetting -> {
+            Feature subsettedFeature = subsetting.getSubsettedFeature();
+            if (subsettedFeature != null) {
+                return subsettedFeature;
+            } else {
+                return subsettedFeature;
+            }
+        };
+    }
+
+    private void appendRedefinition(Appender builder, List<Redefinition> redefinitions, Element element) {
+        if (!redefinitions.isEmpty()) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.REDEFINES);
+            builder.appendSpaceIfNeeded().append(redefinitions.stream().map(Redefinition::getRedefinedFeature)
+                    .filter(Objects::nonNull)
+                    .map(superFeature -> getDeresolvableName(superFeature, element))
+                    .collect(Collectors.joining(", ")));
+        }
+
+    }
+
+    private void appendReferenceSubsetting(Appender builder, ReferenceSubsetting ownedReferenceSubsetting, Element element) {
+        if (ownedReferenceSubsetting != null) {
+            builder.appendSpaceIfNeeded().append(LabelConstants.REFERENCES);
+            if (ownedReferenceSubsetting.getReferencedFeature() != null) {
+                builder.appendSpaceIfNeeded().append(getDeresolvableName(ownedReferenceSubsetting.getReferencedFeature(), element));
+            }
+        }
+
+    }
+
+    /**
+     * @param builder
```

Comment:
Please do not leave method comments like this. Either you really comment a method, or you don't.